src/context.c

```diff
@@ -140,6 +140,51 @@ static gchar* get_cmdline_bootname(void)
 	return bootname;
 }
 
+static gchar* get_custom_bootname(void)
+{
+	g_autoptr(GSubprocessLauncher) launcher = NULL;
+	g_autoptr(GSubprocess) handle = NULL;
+	g_autoptr(GDataInputStream) datainstream = NULL;
+	g_autoptr(GPtrArray) args_array = NULL;
+	g_autofree gchar *outline = NULL;
+	GInputStream *instream;
+	int res;
+
+	args_array = g_ptr_array_new();
+	g_ptr_array_add(args_array, context->config->custom_bootloader_backend);
+	g_ptr_array_add(args_array, (gchar *)("get-current"));
+	g_ptr_array_add(args_array, NULL);
+
+	launcher = g_subprocess_launcher_new(G_SUBPROCESS_FLAGS_STDOUT_PIPE);
+	handle = r_subprocess_launcher_spawnv(launcher, args_array, NULL);
+
+	if (handle == NULL) {
+		g_message("Failed to run custom backend '%s'",
+				context->config->custom_bootloader_backend);
+		return NULL;
+	}
+
+	instream = g_subprocess_get_stdout_pipe(handle);
+	datainstream = g_data_input_stream_new(instream);
+
+	outline = g_data_input_stream_read_line(datainstream, NULL, NULL, NULL);
+
+	res = g_subprocess_wait_check(handle, NULL, NULL);
+	if (!res) {
+		g_message("Failed to get custom backend output");
```

Comment:
Since something went wrong, I would have put a `g_warning()` here. Did you have a reason to prefer `g_message()`?

Reply:
I simply looked at the surrounding code, `get_cmdline_bootname` is doing

`g_message("Failed to resolve realpath for '%s'", bootname);`

But I'm happy to change to g_warning